Userland/Libraries/LibWeb/HTML/Origin.h

```diff
@@ -120,7 +122,10 @@ template<>
 struct Traits<Web::HTML::Origin> : public GenericTraits<Web::HTML::Origin> {
     static unsigned hash(Web::HTML::Origin const& origin)
     {
-        return pair_int_hash(origin.scheme().hash(), pair_int_hash(int_hash(origin.port()), origin.host().hash()));
+        auto hash_without_host = pair_int_hash(origin.scheme().hash(), origin.port());
```

Comment:
You have removed `int_hash(origin.port())`, is this intentional or was there a mistake in the previous code?

Reply:
This line is where the port is being hashed I believe

The change here is mostly to avoid hashing the host when it is null, because it is not supported to serialized a null host

Ideally I think there should be a nice way of hashing this without serialization alltogether though

Reply:
My question revolved around the missing call to `int_hash()` before `origin.port()` is passed onto `pair_int_hash()`, like this:
```suggestion
        auto hash_without_host = pair_int_hash(origin.scheme().hash(), int_hash(origin.port()));
```

Reply:
Ohh I see what you're saying now. No - I didn't notice that we were doing that beforehand, even after you left that comment!

I wonder why we were doing that though? `pair_int_hash` will call `int_hash` on `origin.port()` anyway :thinking: Can't think of how that would meaningfully improve the hashing here

Reply:
Also the value from `origin.scheme().hash()` is rehashed in `pair_int_hash()`. Maybe it doesn't matter, or was it intentionally implemented like this for some reason 🤷🏻‍♂️

Reply:
The rehash of `origin.scheme().hash()` makes sense to me since origin is a `DeprecatedString`, and to be able to hash it in combination with the port/host, need to get an int hash from it. But yeah, for `port`, I don't think it makes a difference since we already have a int for it.

Reply:
Then we can resolve this I think